docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,981 @@
+external_ids        : {"k8s.ovn.org/network"=test12_namespace-scoped, "k8s.ovn.org/topology"=layer2}
+external_ip         : "169.254.0.12"
+external_mac        : []
+external_port_range : ""
+gateway_port        : []
+logical_ip          : "203.203.0.0/16"
+logical_port        : rtos-test12_namespace.scoped_ovn_layer2_switch
+match               : "eth.dst == 0a:58:cb:cb:00:02"
+options             : {stateless="false"}
+priority            : 0
+type                : snat
+```
+
+Also the fact that join IP at gateway router is kept allow to maintain all the NATing done with it at gateway router to define OVN load balancers to implement k8s services.
+
+#### Static Routes and Logical Route Policies
+Changing where routes and policies are configured does not affect TCP connections on upgrade so we can move some routes and policies from gateway router
+to transit_router without issues. In general it will be similar to the layer3 routes and policies.
+
+These will be the routes and policies configured on the new transit_router
+```
+IPv4 Routes
+Route Table <main>:
+                10.96.0.1               203.203.0.2 dst-ip <-- api service
+               10.96.0.10               203.203.0.2 dst-ip <-- dns service
+               100.65.0.2                100.88.0.6 dst-ip <-- node1 traffic use node2 gr peer ip
+               100.65.0.3               100.88.0.10 dst-ip <-- node2 traffic use node2 gr peer ip
+               100.65.0.4               100.88.0.14 dst-ip <-- node3 traffic use node2 gr peer ip
+           203.203.0.0/16               100.88.0.14 src-ip <-- network egress traffic goes to local gateway router peer ip
+```
+```
+      1004 inport == "rtos-test12_namespace.scoped_ovn_layer2_switch" && ip4.dst == 172.18.0.3 /* test12_namespace.scoped_ovn_layer2_switch */         reroute               203.203.0.2
+      1004 inport == "rtos-test12_namespace.scoped_ovn_layer2_switch" && ip6.dst == fc00:f853:ccd:e793::3 /* test12_namespace.scoped_ovn_layer2_switch */         reroute           2010:100:200::2
+       102 (ip4.src == $a10466913729612642039 || ip4.src == $a13607449821398607916) && ip4.dst == $a3613486944346402462           allow
+       102 (ip4.src == $a10466913729612642039 || ip4.src == $a13607449821398607916) && ip4.dst == $a712973235162149816           allow               pkt_mark=1008
+       102 (ip6.src == $a2718358047735721557 || ip6.src == $a13607452020421864338) && ip6.dst == $a1091196985512978262           allow               pkt_mark=1008
+       102 (ip6.src == $a2718358047735721557 || ip6.src == $a13607452020421864338) && ip6.dst == $a3613484745323146040           allow
+       102 ip4.src == 203.203.0.0/16 && ip4.dst == 100.64.0.0/16           allow
+       102 ip4.src == 203.203.0.0/16 && ip4.dst == 203.203.0.0/16           allow
+       102 ip6.src == 2010:100:200::/60 && ip6.dst == 2010:100:200::/60           allow
+       102 ip6.src == 2010:100:200::/60 && ip6.dst == fd98::/64           allow
```

Comment:
and i assume same for all EIP routes? is it that we will create the new ones on transit router and then delete the old ones from GR in phases? or in same transact?

Reply:
different transactions, but that happens after ovn-k restarts with no pods